src/Libraries/Microsoft.Extensions.DataIngestion.MarkItDown/MarkItDownReader.cs

```diff
@@ -0,0 +1,121 @@
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IngestionDocument> ReadAsync(FileInfo source, string identifier, string? mediaType = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(source);
+        _ = Throw.IfNullOrEmpty(identifier);
+
+        if (!source.Exists)
+        {
+            throw new FileNotFoundException("The specified file does not exist.", source.FullName);
+        }
+
+        ProcessStartInfo startInfo = new()
+        {
+            FileName = _exePath,
+            UseShellExecute = false,
+            CreateNoWindow = true,
+            RedirectStandardOutput = true,
+            StandardOutputEncoding = Encoding.UTF8,
+        };
+
+        // Force UTF-8 encoding in the environment (will produce garbage otherwise).
+        startInfo.Environment["PYTHONIOENCODING"] = "utf-8";
+        startInfo.Environment["LC_ALL"] = "C.UTF-8";
+        startInfo.Environment["LANG"] = "C.UTF-8";
+
+#if NET
+        startInfo.ArgumentList.Add(source.FullName);
+        if (_extractImages)
+        {
+            startInfo.ArgumentList.Add("--keep-data-uris");
+        }
+#else
+        startInfo.Arguments = $"\"{source.FullName}\"" + (_extractImages ? " --keep-data-uris" : string.Empty);
+#endif
+
+        string outputContent = string.Empty;
+        using (Process process = new() { StartInfo = startInfo })
+        {
+            process.Start();
+
+            outputContent = await process.StandardOutput.ReadToEndAsync(cancellationToken).ConfigureAwait(false);
+#if NET
+            await process.WaitForExitAsync(cancellationToken).ConfigureAwait(false);
+#else
+            process.WaitForExit();
+#endif
+
+            if (process.ExitCode != 0)
+            {
+                throw new InvalidOperationException($"MarkItDown process failed with exit code {process.ExitCode}.");
+            }
+        }
+
+        return MarkdownParser.Parse(outputContent, identifier);
+    }
+
+    /// <inheritdoc/>
+    /// <remarks>The contents of <paramref name="source"/> are copied to a temporary file.</remarks>
+    public override async Task<IngestionDocument> ReadAsync(Stream source, string identifier, string mediaType, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(source);
+        _ = Throw.IfNullOrEmpty(identifier);
+
+        // Instead of creating a temporary file, we could write to the StandardInput of the process.
+        // MarkItDown says it supports reading from stdin, but it does not work as expected.
+        // Even the sample command line does not work with stdin: "cat example.pdf | markitdown"
+        // I can be doing something wrong, but for now, let's write to a temporary file.
+        string inputFilePath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        using (FileStream inputFile = new(inputFilePath, FileMode.CreateNew, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.Asynchronous))
```

Comment:
This should be placed inside the try, in case the CopyToAsync fails, you still clean up the temporary file.